File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/CommandEncoders/EncoderUtils.java
package bgu.spl171.net.impl.TFTP.CommandEncoders;

public class EncoderUtils {

    static byte[] mergeByteArray(byte[]... arrays) {
        int size = 0;
        for (byte[] array : arrays) {
            size+=array.length;
        }
        byte[] result = new byte[size];

        int index=0;
        for (byte[] array : arrays) {
            System.arraycopy(array, 0, result, index, array.length);
            index+=array.length;
        }
        return result;
    }
}

File: java to c++ encoders/EncoderUtils.cpp
#include "EncoderUtils.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace CommandEncoders
					{

						std::vector<char> EncoderUtils::mergeByteArray(std::vector<char> &arrays)
						{
							int size = 0;
							for (char[] array_Renamed : arrays)
							{
								size += array_Renamed.size();
							}
							std::vector<char> result(size);

							int index = 0;
							for (char[] array_Renamed : arrays)
							{
								System::arraycopy(array_Renamed, 0, result, index, array_Renamed.size());
								index += array_Renamed.size();
							}
							return result;
						}
					}
				}
			}
		}
	}
}
